pkg/destroy/aws/ec2helpers.go

```diff
@@ -296,6 +298,18 @@ func deleteEC2InternetGateway(ctx context.Context, client *ec2.EC2, id string, l
 	return nil
 }
 
+func deleteEC2CarrierGateway(ctx context.Context, client *ec2.EC2, id string, logger logrus.FieldLogger) error {
+	_, err := client.DeleteCarrierGatewayWithContext(ctx, &ec2.DeleteCarrierGatewayInput{
+		CarrierGatewayId: &id,
+	})
+	if err != nil {
+		return err
```

Comment:
Should we check for `NotFound` errors before returning?

Reply:
That's a good point, I didn't found error code for Carrier Gateway in the docs:
https://docs.aws.amazon.com/AWSEC2/latest/APIReference/errors-overview.html#CommonErrors

I am debugging what is returned when CAGW doesn't exist.

Reply:
The installer destroy runs in a loop so I think it would be good to ignore possible `NotFound` errors here instead of failing the whole destroy.

Reply:
@mtulio In a standalone test, I get:
```
Delete carriger gateway: InvalidCarrierGatewayID.NotFound: The CarrierGateway ID 'cagw-0bebd1f0f76ec8fd7' does not exist
	status code: 400, request id: 482e19f0-2074-4db3-9c1f-3e1cb09e36f6
```

Reply:
> of failing the whole destroy.

@r4f4 , apologies, I missed this message. That's a good suggestion. Let me add it. I also submitted a feedback in the docs page.